include/aspect/utilities.h

```diff
@@ -123,6 +123,14 @@ namespace aspect
     }
 
     /**
+     * Given a 2d point and a list of points which form a polygon, computes if the point
+     * falls within the polygon.
+     */
+    template <int dim>
+    bool
+    in_2d_polygon(dealii::Point<2> &point,const std::vector<std::vector<double> > &pointList);
```

Comment:
As a follow-up, reversing the arguments then also suggests naming the function `polygon_contains_point`. No need for the "2d" since a polygon is always 2d (otherwise it'd be a polyhedron).